plugins/modules/system/alternatives.py

```diff
@@ -90,10 +135,210 @@
 class AlternativeState:
     PRESENT = "present"
     SELECTED = "selected"
+    ABSENT = "absent"
+    AUTO = "auto"
 
     @classmethod
     def to_list(cls):
-        return [cls.PRESENT, cls.SELECTED]
+        return [cls.PRESENT, cls.SELECTED, cls.ABSENT, cls.AUTO]
+
+
+class AlternativesModule(object):
+    _UPDATE_ALTERNATIVES = None
+
+    def __init__(self, module):
+        self.module = module
+        self.result = dict(changed=False, diff=dict(before=dict(), after=dict()))
+        self.run()
+
+    @property
+    def mode_present(self):
+        return self.module.params.get('state') in [AlternativeState.PRESENT, AlternativeState.SELECTED, AlternativeState.AUTO]
+
+    @property
+    def mode_selected(self):
+        return self.module.params.get('state') == AlternativeState.SELECTED
+
+    @property
+    def mode_auto(self):
+        return self.module.params.get('state') == AlternativeState.AUTO
+
+    def run(self):
+        self.parse()
+
+        if self.module._diff and self.path in self.current_alternatives:
+            self.result['diff']['before'].update(dict(
+                path=self.path,
+                priority=self.current_alternatives[self.path].get('priority'),
+                link=self.current_link,
+                subcommands=self.current_alternatives[self.path].get('subcommands')
+            ))
+
+        if self.mode_present:
+            # Check if we need to (re)install
+            if (
+                self.path not in self.current_alternatives or
+                self.current_alternatives[self.path].get('priority') != self.priority or
+                not all(s in self.subcommands for s in self.current_alternatives[self.path].get('subcommands')) or
```

Comment:
Is this really needed? Doesn't this imply that if you run the module with `state=present` and no subcommands specified, but subcommands exist, that these subcommands will suddenly be removed?

Reply:
It should be possible to remove some or all sub commands. Subcommands are not left as they are if the subcommands parameter is not set. To remove all `subcommands: []` can be used.

Reply:
I agree, but that didn't work when the default was `[]`. But now you changed it :)